packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -435,6 +430,7 @@ export class AwsCustomResource extends cdk.Construct implements iam.IGrantable {
  */
 function awsSdkToIamAction(service: string, action: string): string {
   const srv = service.toLowerCase();
+  awsSdkMetadata = getAwsSdkMetadata();
```

Comment:
I do not like that the cache variable is written in one function and read in another. Responsibilities should be more clearly divided than that.

Please put the `awsSdkMetadata` global variable close to the `getAwsSdkMetadata()` function in the source, and make sure it's only referenced from that function. If you want to get fancy you can make it a closure, but just co-locating in the source, and maybe renaming to `_awsSdkMetadataCache` or something should inform people no other part of the code should be touching it.

After that, in places where it needs to be consumed (only here?), assing the result to a local variable and use that.